src/superannotate/lib/core/usecases.py

```diff
@@ -3249,56 +3249,57 @@ def execute(self):
+            response = self._backend_service.get_annotation_upload_data(
                 project_id=self._project.uuid,
                 team_id=self._project.team_id,
                 folder_id=self._folder.uuid,
                 image_ids=[image_data["id"]],
             )
-            session = boto3.Session(
-                aws_access_key_id=auth_data["creds"]["accessKeyId"],
-                aws_secret_access_key=auth_data["creds"]["secretAccessKey"],
-                aws_session_token=auth_data["creds"]["sessionToken"],
-                region_name=auth_data["creds"]["region"],
-            )
-            resource = session.resource("s3")
-            bucket = resource.Bucket(auth_data["creds"]["bucket"])
-            self.fill_classes_data(self._annotations)
-            bucket.put_object(
-                Key=auth_data["images"][str(image_data["id"])]["annotation_json_path"],
-                Body=json.dumps(self._annotations),
-            )
-            if (
-                self._project.project_type == constances.ProjectType.PIXEL.value
-                and os.path.exists(self._annotation_path)
-            ):
-                with open(self._annotation_path, "rb") as descriptor:
-                    file = io.BytesIO(descriptor.read())
+            if response.ok:
+                session = boto3.Session(
+                    aws_access_key_id=response.data.access_key,
+                    aws_secret_access_key=response.data.secret_key,
+                    aws_session_token=response.data.session_token,
+                    region_name=response.data.region,
+                )
+                resource = session.resource("s3")
+                bucket = resource.Bucket(response.data.bucket)
+                self.fill_classes_data(self._annotations)
+                bucket.put_object(
+                    Key=response.data.images[image_data["id"]]["annotation_json_path"],
+                    Body=json.dumps(self._annotations),
+                )
+                if (
+                    self._project.project_type == constances.ProjectType.PIXEL.value
+                    and os.path.exists(self._annotation_path)
+                ):
+                    with open(self._annotation_path, "rb") as descriptor:
+                        file = io.BytesIO(descriptor.read())
+                        bucket.put_object(
+                            Key=response.data.images[image_data["id"]][
+                                "annotation_bluemap_path"
+                            ],
+                            Body=file,
+                        )
+                if self._verbose:
+                    logger.info(
+                        "Uploading annotations for image %s in project %s.",
+                        str(image_data["name"]),
+                        self._project.name,
+                    )
+
+                if (
+                    self._project.project_type == constances.ProjectType.PIXEL.value
```

Comment:
This can be in line 3273

Reply:
This very confusing case, thanks updated